styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorageTransaction.java

```diff
@@ -173,6 +174,28 @@ public WorkflowId store(Workflow workflow) throws IOException {
     return workflow.id();
   }
 
+  @Override
+  public WorkflowId storeWorkflowWithNextNaturalTrigger(Workflow workflow, TriggerInstantSpec triggerSpec)
+      throws IOException {
+
+    final Key componentKey = DatastoreStorage.componentKey(tx.getDatastore().newKeyFactory(), workflow.componentId());
+    if (tx.get(componentKey) == null) {
```

Comment:
This would make https://github.com/spotify/styx/pull/485 incomplete, but anyway.